src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/VsSolutionEventListener.cs

```diff
@@ -152,11 +159,15 @@ public int PrioritizedOnBeforeUnloadProject(IVsHierarchy pRealHierarchy, IVsHier
 
         public int PrioritizedOnAfterOpenSolution(object pUnkReserved, int fNewSolution)
         {
+            IsSolutionClosing = false;
+
             return HResult.NotImplemented;
         }
 
         public int PrioritizedOnBeforeCloseSolution(object pUnkReserved)
         {
+            IsSolutionClosing = true;
```

Comment:
Pick Prioritized or not-Prioritized, not both. They are the same events just one (Prioritized) fires earlier than the other.

Reply:
> just one (Prioritized) fires earlier than the other

I thought the same, and it's true during load, but during unload/close the order is reversed.

It's probably still safe to only used the prioritized methods, but I think the current approach is also safe so long as load/unload cannot overlap (which seems like a reasonable assumption).